tests/ssl-opt.sh

```diff
@@ -483,10 +483,16 @@ detect_required_features() {
             requires_config_enabled MBEDTLS_SSL_ALPN;;
     esac
 
+    case " $CMD_LINE " in
+         *\ auth_mode=*|*[-_\ =]crt[_=]*)
```

Comment:
In practice it matches Mbed TLS's `crt_file` and `context_crt_cb` options. It's a copy of the pattern used in `maybe_adapt_for_psk` introduced in https://github.com/Mbed-TLS/mbedtls/pull/5582. The pattern is intended to catch test cases that intrinsically depend on using a certificate. (Future-proofing for the hypothetical time we add support for raw public keys is out of scope.)

This is a heuristic intended to catch current cases as well as likely future cases. I can change it if you think it's wrong, but it seems out of scope since the current heuristic isn't causing any test case failure, nor any unwanted skip that I am aware of.

Reply:
What about adding a comment explaining that?